faq/OPTIONS.md

```diff
@@ -0,0 +1,29 @@
+### Pass Rollup Output Configuration Directly to Plugin
+
+This plugin transforms JavaScript input to achieve maximum compression. It is necessary to know the intended output format of your code when transformations are run on the input source.
+
+Unfortunately, the Rollup API doesn't expose these options until later lifecycle hooks.
```

Comment:
There is `options` hook, but unfortunately it receives only `inputOptions` as argument. Knowing output options seems like generally useful stuff, could be added to rollup. WDYT @lukastaegert @guybedford ?

Reply:
I agree, it would be great for Rollup to provide the `OutputOptions` with the options callback. Right now, without it users must supply the configuration twice.

Reply:
I was able to refactor the implementation and inject `OutputOptions` to each of the transforms only once it was known.

I still believe it would be good to know this earlier, but have worked around it for now.

Reply:
Hi, as I am still on vacation in Italy just a short heads-up as I was reading through this. Unfortunately it is impossible to provide the output options earlier than the transformChunk hook as they might just not be known earlier. Slightly simplified, Rollup’s bundling has two phases that can both be triggered manually through the node API. 

The first “rollup” phase loads and parses the modules, runs the tree-shaking and creates a format-less “raw bundle”. It is during that phase that the “transform” hook is run. This phase solely depends on the input options.

Then in the “generate” phase you can finalize as many bundles as you want from the raw bundle. Each set of output options corresponds to one finalised bundle which is why you get the output options in the “transformChunk” hook that is run during that phase. Usually, a minifier should be applied with this hook for optimal results (rather than the transform hook), but I did not dig too deeply into the ideas behind this plugin. If you pass an array of output options in your config file, the CLI interface will automatically run “generate” once for each output so you can be sure that the transformChunk hook does not receive an array of options.

As you cannot be sure if and how “generate” will be run during the “rollup” phase, you can thus not access the output options.

Hope this helps a little.

Reply:
Quite helpful @lukastaegert. Enjoy your vacation and we can catch up when you return.